cmd/snap-failure/cmd_snapd.go

```diff
@@ -145,6 +145,15 @@ func (c *cmdSnapd) Execute(args []string) error {
 	}
 
 	logger.Noticef("restoring invoking snapd from: %v", snapdPath)
+	if prevRev != "0" {
+		currentSymlink := filepath.Join(dirs.SnapMountDir, "snapd", "current")
+		if err := os.Remove(currentSymlink); err != nil && !os.IsNotExist(err) {
+			return fmt.Errorf("cannot remove symlink %s: %v", currentSymlink, err)
+		}
+		if err := os.Symlink(prevRev, currentSymlink); err != nil {
+			return fmt.Errorf("cannot create symlink %s: %v", currentSymlink, err)
+		}
```

Comment:
should we use osutil.AtomicSymlink for this instead?

Reply:
should tests in the package check for this step?

Reply:
I have realized I forgot to do this one yesterday. I will do it when I am back.

Reply:
I have switched to AtomicSymlink and added a test.